### PR TITLE
Update Gatsby to use Babel 6

### LINKED ISSUE #195: Getting error with babel inside Gatsby
Everytime I try to do `gatsby develop` I get this error.

I've tried the two latest versions of gatsby with the same result.
I tried use the master from github, ran the build scripts, set published to true and then got the same result.

I guess this is connected to the other issue to upgrade to babel 6. But am I just missing something?

```sh
/usr/local/lib/node_modules/gatsby/node_modules/babel-core/lib/transformation/file/options/option-manager.js:126
      if (!option) this.log.error("Unknown option: " + alias + "." + key, ReferenceError);
                           ^

TypeError: Cannot read property 'error' of undefined
  at OptionManager.mergeOptions (/usr/local/lib/node_modules/gatsby/node_modules/babel-core/lib/transformation/file/options/option-manager.js:126:28)
  at OptionManager.addConfig (/usr/local/lib/node_modules/gatsby/node_modules/babel-core/lib/transformation/file/options/option-manager.js:107:10)
  at OptionManager.findConfigs (/usr/local/lib/node_modules/gatsby/node_modules/babel-core/lib/transformation/file/options/option-manager.js:168:35)
  at OptionManager.init (/usr/local/lib/node_modules/gatsby/node_modules/babel-core/lib/transformation/file/options/option-manager.js:229:12)
  at compile (/usr/local/lib/node_modules/gatsby/node_modules/babel-core/lib/api/register/node.js:117:22)
  at normalLoader (/usr/local/lib/node_modules/gatsby/node_modules/babel-core/lib/api/register/node.js:199:14)
  at Object.require.extensions.(anonymous function) [as .js] (/usr/local/lib/node_modules/gatsby/node_modules/babel-core/lib/api/register/node.js:216:7)
  at Module.load (module.js:356:32)
  at Function.Module._load (module.js:311:12)
  at Module.require (module.js:366:17)
  at require (module.js:385:17)
  at module.exports (/usr/local/lib/node_modules/gatsby/dist/utils/glob-pages.js:42:13)
  at module.exports (/usr/local/lib/node_modules/gatsby/dist/utils/develop.js:85:37)
  at Object.<anonymous> (/usr/local/lib/node_modules/gatsby/bin/develop.js:36:1)
  at Module._compile (module.js:435:26)
  at Module._extensions..js (module.js:442:10)
  at Object.require.extensions.(anonymous function) [as .js] (/usr/local/lib/node_modules/gatsby/node_modules/babel-core/lib/api/register/node.js:214:7)
  at Module.load (module.js:356:32)
  at Function.Module._load (module.js:311:12)
  at Module.require (module.js:366:17)
  at require (module.js:385:17)
  at Object.<anonymous> (/usr/local/lib/node_modules/gatsby/bin/gatsby-develop.js:4:1)
  at Module._compile (module.js:435:26)
  at Object.Module._extensions..js (module.js:442:10)
  at Module.load (module.js:356:32)
  at Function.Module._load (module.js:311:12)
  at Function.Module.runMain (module.js:467:10)
  at startup (node.js:136:18)
  at node.js:963:3
```

### PR DESCRIPTION
**Pull request by KyleAMathews**
Fixes #195 #153

### CODE REVIEW

**Comment by tleunen** [comment]
Installed this branch both locally and globally, and I get this error:
```
~/d/r/ndocs gatsby ❯ gatsby develop                                                                                                   ✭
/Users/tleunen/.nvm/versions/node/v5.9.0/lib/node_modules/gatsby/lib/utils/develop.js:2
import Hapi from 'hapi'
^^^^^^

SyntaxError: Unexpected token import
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:387:25)
    at Module._extensions..js (module.js:422:10)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/tleunen/.nvm/versions/node/v5.9.0/lib/node_modules/gatsby/node_modules/babel-register/lib/node.js:134:7)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/Users/tleunen/.nvm/versions/node/v5.9.0/lib/node_modules/gatsby/bin/develop.js:14:13)
    at Module._compile (module.js:413:34)
```

I get the same thing when running locally, with `../node_modules/.bin/gatsby develop`

**Comment by KyleAMathews** [comment]
To test his manually you need to first compile the code in `/lib` to `/dist` and second change `/bin/published.js` to true from false.

**Comment by tleunen** [comment]
I was just running `build.sh`. I changed the value to false in prepublish and ran it... Same issue, when doing it locally.
In the global gatsby, I'm not even able to finish the `npm install`. I get this weird error: `npm ERR! Cannot read property 'target' of null`

**Comment by tleunen** [comment]
Ok, I took a look at the code, and I only decided to run `build.sh` and use the dist files instead of the lib files. Therefore I didn't change the value in `published.js`.
I wasn't able to use the source files, it seems there's still an issue with the babel config for that.

File: lib/utils/develop.js [hunk 1]
```diff
@@ -46,13 +36,17 @@ module.exports = (program) => {
       test: /\.jsx?$/, // Accept either .js or .jsx files.
       exclude: babelExcludeTest,
```
Comment (tleunen) [inline lib/utils/develop.js]: You removed `utils/babel-exclude-test.js`, so you have remove the import in this file and replace this with `exclude: /(node_modules|bower_components)/,`

**Comment by tleunen** [comment]
After the import issue, I was able to run the `develop` command, with the "published" code.

**Comment by KyleAMathews** [comment]
@tleunen thanks for your testing! The issues you raised should be fixed now.

**Comment by tleunen** [comment]
I got this postcss warning after running the build process.
> Warning: postcss-cssnext found a duplicate plugin ('autoprefixer') in your postcss plugins. This might be inefficient. You should remove 'autoprefixer' from your postcss plugin list since it's already included by postcss-cssnext.

I'm not using it in my project, so I guess it only comes from gatsby :)

**Comment by cutemachine** [comment]
Yes, I see this too.
Tommy <notifications@github.com> schrieb am Di., 22. März 2016 um 03:47:

**Comment by KyleAMathews** [comment]
Yeah, saw that too (PostCSS message). Not sure why, only showed up after Babel 6 upgrade I think. If either of you want to debug how it's getting added twice, feel free :)

**Pull request merged**